packages/astro/components/ViewTransitions.astro

```diff
@@ -0,0 +1,137 @@
+---
+type Fallback = 'none' | 'animate' | 'swap';
+
+export interface Props {
+	fallback?: Fallback;
+}
+
+const { fallback = 'animate' } = Astro.props as Props;
+---
+<meta name="astro-view-transitions-enabled" content="true">
+<meta name="astro-view-transitions-fallback" content={fallback}>
+<script>
+	type Fallback = 'none' | 'animate' | 'swap';
+	type Direction = 'forward' | 'back';
+
+	let currentIndex = history.state?.index || 0;
+	if(!history.state) {
+		history.replaceState({index: currentIndex}, document.title);
+	}
+
+	const supportsViewTransitions = !!document.startViewTransition;
+	const transitionEnabledOnThisPage = () => !!document.querySelector('[name="astro-view-transitions-enabled"]');
+
+  async function getHTML(href: string) {
+    let res = await fetch(href)
+    let html = await res.text();
+		return { ok: res.ok, html };
+  }
+
+	function getFallback(): Fallback {
+		let el = document.querySelector('[name="astro-view-transitions-fallback"]');
+		if(el) {
+			return el.getAttribute('content') as Fallback;
+		}
+		return 'animate';
+	}
+
+	const parser = new DOMParser();
+
+  async function updateDOM(dir: Direction, html: string, fallback: Fallback = 'swap') {
```

Comment:
Here the default `fallback` is `swap`, but we return `'animate'` inside the function `getFallback`. Is it intended?

Reply:
This was really a weird case to handle non-fallback. I made the argument optional instead.